tests/components/telegram_bot/conftest.py

```diff
@@ -65,6 +65,23 @@ def mock_register_webhook():
         yield
 
 
+@pytest.fixture
+def mock_generate_secret_token():
+    """Mock secret token generated for webhook"""
+    mock_secret_token = "DEADBEEF12345678DEADBEEF87654321"
+    with patch(
+        "homeassistant.components.telegram_bot.webhooks.generate_secret_token",
+        return_value=mock_secret_token,
```

Comment:
We should not patch the code we're testing.
Instead, do like this, calls to `secrets.choice` will now iterate over `mock_secret_token`:
```suggestion
        "homeassistant.components.telegram_bot.webhooks.secrets.choice",
        side_effect=mock_secret_token,

```